### PR TITLE
added usb-iso.sh to create a usb key that can be used to install nodes at remote site where pxe is not feasible

### PR DESCRIPTION
**Pull request by jianzzha**
This purpose of this script can be found here:
https://docs.google.com/document/d/12tfUCscYhC1hMBqh2aXjz688oRktZ2r0Nr9kXSq-J5A/edit?usp=sharing

### CODE REVIEW

File: hack/usb-iso.sh [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+parse_args() {
+   WRITE_USB=${WRITE_USB:-true}
+
+   USAGE="Usage: $0 [options]
+Options:
+    -d USB_DEV     Install to the given usb device, alternatively
+                   specify env var USB_DEV
+    -i IGNITION    The URL or path to the given Ignition config,
```
Comment (yrobla) [inline hack/usb-iso.sh:10]: we need some feature, to indicate if we simply want to inject the URL, or we want to downloaded the generated ignition adn embed it into the initramfsExtra image

File: hack/usb-iso.sh [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	echo "usb device must be specified unless WRITE_USB is set to false!"
+	echo "$USAGE"
+	exit 1
+    fi
+    # normalize device name
+    USB_DEV=$(echo ${USB_DEV} | sed -r 's|/dev/||')
+    USB_DEV=/dev/${USB_DEV}
+
+    if [[ -z "${IGNITION_URL}" ]]; then
+	echo "ignition url must be specified!"
```
Comment (yrobla) [inline hack/usb-iso.sh:46]: if we are just passing an URL there, and not downloading the file, this validation is not needed

File: hack/usb-iso.sh [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    mount -t iso9660 -o loop installer.iso _iso_mount 
+
+    echo "copy out installer files ..."
+    pushd _iso_mount 
+    tar cf - . | (cd ../_rhcos_custom && tar xfp -)
+    popd
+
+    echo "update installer parameters ..."
+    pushd _rhcos_custom 
+    sed -i -r "s|^(  append initrd)=.*|\1=/images/initramfs.img,/images/initramfsExtra nomodeset ip=dhcp rd.neednet=1 coreos.inst=yes console=tty0 console=ttyS0 coreos.inst.ignition_url=file:///extra/node.ign coreos.inst.image_url=file:///extra/rhcos.raw.gz coreos.inst.install_dev=${BLOCK_DEV}|g" isolinux/isolinux.cfg
```
Comment (yrobla) [inline hack/usb-iso.sh:121]: file:///extra/node.ign could be just ${IGNITION_URL} depending on the use case

File: hack/usb-iso.sh [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    popd
+
+    echo "building extra initramfs ..."
+    pushd _extra
+    find . | sed 's/^[.]\///' | cpio -o -H newc --no-absolute-filenames > ../_rhcos_custom/images/initramfsExtra
+    popd
+   
+    echo "rebuilding installer image ..."
+    pushd _rhcos_custom
+    mkisofs -o rhcos_custom_usb.iso -b isolinux/isolinux.bin -c isolinux/boot.cat  -no-emul-boot -boot-load-size 4 -boot-info-table -R -J -V "RHCOS custom installer" .
```
Comment (yrobla) [inline hack/usb-iso.sh:131]: when burning to an usb, we need:
isohybrid rhcos_custom_usb.iso

After the iso creation, so it generates the proper structure for the usb to boot.

File: hack/usb-iso.sh [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    echo "writing usb disk ..."
+    pushd _rhcos_custom
+    dd bs=4M if=rhcos_custom_usb.iso of=${USB_DEV} status=progress oflag=sync
+    popd
+
+    echo "cleaning up ..."
+    umount _iso_mount
+    popd
+    /bin/rm -f _data
```
Comment (yrobla) [inline hack/usb-iso.sh:148]: this is a directory, right? needs to be rm -rf _data

File: hack/usb-iso.sh [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+Options:
+    -d USB_DEV     Install to the given usb device, alternatively
+                   specify env var USB_DEV
+    -i IGNITION    The URL or path to the given Ignition config,
+                   alternatively specify env var IGNITION_URL
+    -o INSTALLER   The URL or path to the installer image,
+                   alternatively specify env var INSTALLER_IMAGE
+    -m METAL_IMG   The URL or path to the Metal image, 
+                   alternatively specify env var METAL_IMAGE
+    -b BLOCK_DEV   The block device on the remote node,
```
Comment (yrobla) [inline hack/usb-iso.sh:16]: This needs to be just sda, as coreos-installer is adding the /dev/ prefix

**Comment by jianzzha** [comment]
@patrickdillon @mtnbikenc This PR is to meet a telcom requirement (document link in top comments). Can you ok-to-test so we can get it complete?

**Pull request closed without merging**